pos-print/src/main/java/com/clouway/pos/print/adapter/http/PrintService.kt

```diff
@@ -25,22 +26,32 @@ class PrintService @Inject constructor(private var factory: PrinterFactory) {
     try {
       val dto = request.read(ReceiptDTO::class.java).`as`(GsonTransport::class.java)
       val printer = factory.getPrinter(dto.sourceIp)
-      when {
-        dto.fiscal
-        -> response = printer.printFiscalReceipt(dto.receipt)
-        else
-        -> response = printer.printReceipt(dto.receipt)
+      try {
+        when {
+          dto.fiscal
+          -> response = printer.printFiscalReceipt(dto.receipt)
+          else
+          -> response = printer.printReceipt(dto.receipt)
+        }
+      } catch (e: RequestTimeoutException) {
+        return Reply.with(ErrorResponse("Printer request timeout.\n" + e.message)).`as`(GsonTransport::class.java).status(504)
+      } finally {
+        printer.close()
       }
-      printer.close()
     } catch (e: DeviceNotFoundException) {
-      return Reply.with(ErrorResponse("Device not found.")).badRequest()
+      return Reply.with(ErrorResponse("Device not found.")).`as`(GsonTransport::class.java).notFound()
     } catch (e: IOException) {
-      return Reply.with(ErrorResponse("Device can't connect.")).status(480)
+      return Reply.with(ErrorResponse("Device can't connect.")).`as`(GsonTransport::class.java).status(480)
     }
+
     val responseDTO: PrintReceiptResponseDTO = PrintReceiptResponseDTO(response.warnings)
-    return Reply.with(responseDTO).`as`(GsonTransport::class.java)
-  }
+    println(GsonBuilder().create().toJson(responseDTO))
 
+    return if (responseDTO.warnings.contains(Status.FISCAL_RECEIPT_IS_OPEN) ||
+      responseDTO.warnings.contains(Status.NON_FISCAL_RECEIPT_IS_OPEN)) {
+      Reply.with(responseDTO).`as`(GsonTransport::class.java)
+    } else Reply.with(ErrorResponse("Didn't open receipt.")).`as`(GsonTransport::class.java).badRequest()
```

Comment:
What does this means ?  is this an error? I thnk that there should be a connection or some printer problem if an receipt cannot be opened. I think that The client app don't care if the receipt can be open or not!. The client cares if the receipt is print or not and if there are some problems with the device(or configurations)
the decsription is not giving me actual rason and a result (it is 400  - so it is error)